lib/postzord/receiver/local_batch.rb

```diff
@@ -25,6 +25,15 @@ def receive!
     # 09/27/11 this is slow
     notify_users
 
+    if @object.respond_to?(:type) && @object.type == "Reshare"
+      root_object = Post.find_by guid: @object.root_guid
+      root_author = Person.find(root_object.author_id)
+      if root_author.local?
+        # create a participation to the user for this reshare, to federate comments to author
+        root_author.owner.participate! @object
+      end
+    end
+
```

Comment:
On second though, how about moving the logic [here](https://github.com/diaspora/diaspora/blob/develop/app/models/reshare.rb#L59)?